File: study/lua/src/app/scene/view/petHandBook/PetAvatarNode.lua
--神兽avatar 展示
local ViewBase = require("app.ui.ViewBase")
local PetAvatarNode = class("PetAvatarNode", ViewBase)

local TypeConvertHelper = require("app.utils.TypeConvertHelper")
local UserDataHelper = require("app.utils.UserDataHelper")
local UIPopupHelper	 = require("app.utils.UIPopupHelper")



function PetAvatarNode:ctor()
    self._commonAvatar = nil
    self._textPetName  = nil
    self._imagePetColor = nil
    self._imageNoShow = nil
    local resource = {
        file = Path.getCSB("PetAvatarNode", "pet"),
    }
    PetAvatarNode.super.ctor(self, resource)
end

function PetAvatarNode:updateUI(petId)
    local params = TypeConvertHelper.convert(TypeConvertHelper.TYPE_PET,petId)

    self:updateBaseId(petId)

end

function PetAvatarNode:doNoShow( ... )
    -- body
    self._imageNoShow:setVisible(true)
    self._commonAvatar:setVisible(false)
end
function PetAvatarNode:darkNode( ... )
    -- body
    self._imageNoShow:setVisible(false)
    self._commonAvatar:setVisible(true)
    self._commonAvatar:setAction("idle_ex",true)
end

function PetAvatarNode:highLightNode( ... )
    -- body
    self._imageNoShow:setVisible(false)
    self._commonAvatar:setVisible(true)
    self._commonAvatar:setAction("idle",true)
   
end

function PetAvatarNode:onCreate()
     self._commonAvatar:setTouchEnabled(false)
end

function PetAvatarNode:turnBack()
    self._commonAvatar:turnBack()
end

function PetAvatarNode:updateBaseId(baseId,animSuffix)
    self._commonAvatar:setConvertType(TypeConvertHelper.TYPE_PET)
    self._baseId = baseId
    self._commonAvatar:updateUI(baseId,"_small")
    self._commonAvatar:setScale(1.6)
end

function PetAvatarNode:setAction(ani,loop)
    self._commonAvatar:setAction(ani,loop)
end

function PetAvatarNode:setCallBack(callBack)
    self._commonAvatar:setTouchEnabled(true)
    self._commonAvatar:setCallBack(callBack)
end

function PetAvatarNode:setUserData( data )
    -- body
    self._commonAvatar:setUserData(data)
end

--
function PetAvatarNode:onEnter()

end

function PetAvatarNode:onExit()

end

return PetAvatarNode

File: study/ts/script/scene/view/petHandBook/PetAvatarNode.ts
const { ccclass, property } = cc._decorator;

import CommonHeroAvatar from '../../../ui/component/CommonHeroAvatar'
import { TypeConvertHelper } from '../../../utils/TypeConvertHelper';

@ccclass
export default class PetAvatarNode extends cc.Component {

    @property({
        type: CommonHeroAvatar,
        visible: true
    })
    _commonAvatar: CommonHeroAvatar = null;

    @property({
        type: cc.Sprite,
        visible: true
    })
    _imageNoShow: cc.Sprite = null;

    @property({
        type: cc.Node,
        visible: true
    })
    _clickPanel: cc.Node = null;
    _baseId: any;

    init() {
        this._commonAvatar.init();
    }

    updateUI(petId) {
        var params = TypeConvertHelper.convert(TypeConvertHelper.TYPE_PET, petId);
        this.updateBaseId(petId);
    }
    doNoShow() {
        this._imageNoShow.node.active = (true);
        this._commonAvatar.node.active = (false);
    }
    darkNode() {
        this._imageNoShow.node.active = (false);
        this._commonAvatar.node.active = (true);
        this._commonAvatar.setAction('idle_ex', true);
    }
    highLightNode() {
        this._imageNoShow.node.active = (false);
        this._commonAvatar.node.active = (true);
        this._commonAvatar.setAction('idle', true);
    }
    onCreate() {
        this._commonAvatar.setTouchEnabled(false);
    }
    turnBack() {
        this._commonAvatar.turnBack();
    }
    updateBaseId(baseId, animSuffix?) {
        this._commonAvatar.setConvertType(TypeConvertHelper.TYPE_PET);
        this._baseId = baseId;
        this._commonAvatar.updateUI(baseId, '_small');
        this._commonAvatar.setScale(1.6);
    }
    setAction(ani, loop) {
        this._commonAvatar.setAction(ani, loop);
    }
    setCallBack(callBack) {
        this._commonAvatar.setTouchEnabled(true);
        this._commonAvatar.setCallBack(callBack);
    }
    setUserData(data) {
        this._commonAvatar.setUserData(data);
    }
    onEnter() {
    }
    onExit() {
    }

}